pkg/api/entries.go

```diff
@@ -192,12 +200,101 @@ func GetLogEntryByIndexHandler(params entries.GetLogEntryByIndexParams) middlewa
 	return entries.NewGetLogEntryByIndexOK().WithPayload(logEntry)
 }
 
+func getArtifactHashValue(entry types.EntryImpl) crypto.Hash {
+	artifactHash, err := entry.ArtifactHash()
+	if err != nil {
+		// Default to SHA256 if no artifact hash is specified
+		return crypto.SHA256
+	}
+
+	var artifactHashAlgorithm string
+	algoPosition := strings.Index(artifactHash, ":")
+	if algoPosition != -1 {
+		artifactHashAlgorithm = artifactHash[:algoPosition]
+	}
+	switch artifactHashAlgorithm {
+	case "sha256":
+		return crypto.SHA256
+	case "sha384":
+		return crypto.SHA384
+	case "sha512":
+		return crypto.SHA512
+	default:
+		return crypto.SHA256
+	}
+}
+
+func getPublicKey(identity identity.Identity) (crypto.PublicKey, error) {
+	switch identityCrypto := identity.Crypto.(type) {
+	case *x509.Certificate:
+		return identityCrypto.PublicKey, nil
+	case *rsa.PublicKey:
+		return identityCrypto, nil
+	case *ecdsa.PublicKey:
+		return identityCrypto, nil
+	case ed25519.PublicKey:
+		return identityCrypto, nil
+	default:
+		return nil, fmt.Errorf("unsupported public key type: %T", identityCrypto)
+	}
+}
+
+func checkEntryAlgorithms(entry types.EntryImpl) (bool, error) {
+	// Only check algorithms for hashedrekord entries
+	switch entry.(type) {
+	case *hashedrekord.V001Entry:
+		break
+	default:
+		return true, nil
+	}
+
```

Comment:
@bobcallaway @haydentherapper shall we start with hashedrekord entries only? If we want to add support for other types I guess we need to change the schemas similarly to what we did for hashedrekord.

Reply:
My 0.02c is that it's fine to start here: my understanding is that the long term plan is to remove a lot of the non-hashed entries with Rekor v2 anyways, plus this flag is primarily for test deployment purposes anyways 🙂.

Reply:
late reply, but yes, I agree, starting with just hashedrekord is totally reasonable